runtime/src/accounts_db.rs

```diff
@@ -4402,6 +4402,23 @@ impl AccountsDb {
         }
     }
 
+    pub fn hash_account_with_rent_epoch<T: ReadableAccount>(
+        slot: Slot,
+        account: &T,
+        pubkey: &Pubkey,
+        rent_epoch: Epoch,
+    ) -> Hash {
+        Self::hash_account_data(
+            slot,
+            account.lamports(),
+            account.owner(),
+            account.executable(),
+            rent_epoch,
+            account.data(),
+            pubkey,
+        )
+    }
+
     pub fn hash_account<T: ReadableAccount>(slot: Slot, account: &T, pubkey: &Pubkey) -> Hash {
```

Comment:
I considered having this function call `hash_account_with_rent_epoch(..., account.rent_epoch())`, but I'm unsure if the extra parameter wrangling on the stack(?) would introduce a slower code path. Hashing accounts is already a hot point in replay and block processing. Maybe this would be irrelevantly slower or maybe optimizer would inline it all. The goal would be to eliminate duplicate code to make the code simpler and easier to test.

Reply:
https://stackoverflow.com/questions/37639276/when-should-inline-be-used-in-rust

I think we will be fine here - this two functions are in the same module, rust compiler should be able to choose the inline for use when -O flag is on. And the function overhead is negligible compare with the hash in the actual function.